### PR TITLE
Allowed reupload of same module if already installed

### PR DESCRIPTION
**Pull request by mickaelandrieu**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | Before, reinstall an already installed module using a zip file produced an error. Now, the previous module is uninstalled and the new one is installed. |
| Type? | bug fix |
| Category? | BO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-637 |
| How to test? | Try to install an already installed module using Zip/archive method |

#### Important guidelines
- Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
- Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!
- Your commit name MUST respect our [naming convention](http://doc.prestashop.com/display/PS16/How+to+write+a+commit+message)!

### CODE REVIEW

**Comment by kpodemski** [comment]
If you are uploading the same module it should not be uninstalled, it should be updated or leave as it was before. This is how it's working now

**Comment by mickaelandrieu** [comment]
@kpodemski I know but update for zip files isnt implemented for now.

See http://forge.prestashop.com/browse/BOOM-1054

**Comment by kpodemski** [comment]
Well, in that case maybe simple warning that module is already available - of course if there's no update files.

Could you assure everyone that update via zip upload will be available in 1.7.0?

File: src/Core/Addon/Module/ModuleManager.php [hunk 1]
```diff
@@ -104,12 +104,8 @@ public function install($name)
...
 
         if ($this->moduleProvider->isInstalled($name)) {
-            throw new Exception(
-                $this->translator->trans(
-                    'The module %module% is already installed.',
-                    array('%module%' => $name),
-                    'Admin.Modules.Notification'
-                ));
+            $module = $this->moduleRepository->getModule($name);
+            $module->onUninstall();
```
Comment (Quetzacoalt91) [inline src/Core/Addon/Module/ModuleManager.php]: No no no ! Do not uninstall the module during an upgrade ! It may delete some data.

Comment (kpodemski) [inline src/Core/Addon/Module/ModuleManager.php]: Yay, voice of reason!

Comment (mickaelandrieu) [inline src/Core/Addon/Module/ModuleManager.php]: @Quetzacoalt91 ok. So... there is no implemented method to make an upgrade both in Adapter model and in ModuleManager.

**Comment by mickaelandrieu** [comment]
As I've already said, the issue is not an upload issue, the main issue is that we can't upgrade any module using the Zip/Archive method for now (See http://forge.prestashop.com/browse/BOOM-1054).

So this contribution is useless atm ;)

**Pull request closed without merging**